src/CTA.Rules.Actions/InvocationExpressionActions.cs

```diff
@@ -23,6 +23,17 @@ InvocationExpressionSyntax ReplaceMethod(SyntaxGenerator syntaxGenerator, Invoca
             return ReplaceMethod;
         }
 
+        public Func<SyntaxGenerator, InvocationExpressionSyntax, InvocationExpressionSyntax> GetReplaceOnlyMethodAction(string oldMethod, string newMethod)
+        {
+            //TODO what's the outcome if newMethod doesn't have a valid signature.. are there any options we could provide to parseexpression ?
+            InvocationExpressionSyntax ReplaceOnlyMethod(SyntaxGenerator syntaxGenerator, InvocationExpressionSyntax node)
+            {
+                node = node.WithExpression(SyntaxFactory.ParseExpression(node.Expression.ToString().Replace(oldMethod, newMethod))).NormalizeWhitespace();
+                return node;
+            }
+            return ReplaceOnlyMethod;
+        }
```

Comment:
Another concern- it looks like the method is being replaced using string replacement, which may have unintended consequences. Hypothetical example:
```csharp
// Initial node
Add(AdditionProperty1, AdditionProperty2);

// Result after replacing "Add" method with "Subtract"
Subtract(SubtractitionProperty1, SubtractitionProperty2);
```

Is there a more robust way to reconstruct the invocation expression, explicitly preserving the parameters?

Reply:
I definitely agree with avoiding confusion and adding comments and examples in there. Updated as you proposed.

Reply:
As for the second point, I completely agree with the concerns here. I spent a lot of time trying to figure out how to do it with proper syntaxfactory objects but the invocation expression just doesn't seem to be able to access that component.

I can see it via node.expression.name.identifier but that is not accessible via object accessors. There is no expression named name. This was the only way i could figure out how to do it but i know its prone to error. If there are any better suggestions I am definitely willing to use a better way.

Reply:
Comments look good, thanks for adding those. As we discussed offline, there is a way to create a new InvocationExpression while preserving the arguments, but it does erase any prefixes or qualifiers. 

Example:
```csharp
// This creates a new invocation expression with existing parameters
node = SyntaxFactory.InvocationExpression(
        SyntaxFactory.IdentifierName(newMethod), 
        node.ArgumentList)
    .NormalizeWhitespace()

// Example Results    
// Original node: Console.WriteLine("SomeString")
// New method: NewMethod

// Desired output: Console.NewMethod("SomeString")
// Actual output: NewMethod("SomeString")    
```

Reply:
+1 to Jon's comment. I suggest the below:
1) ReplaceMethodWithObjectAndParameters -> Replaces the full method call, object, and parameters
2) ReplaceMethodWithObject -> Replaces the fullmethod call and object, keeping the parameters
3) ReplaceMethodAndParameters -> Replaces the method call and parameters
4) ReplaceMethodOnly -> Replaces the method call only

We might need all scenarios because some method replacements will not have the same parameters. Let me know what you think.